### DISCUSSION TITLE
Zigbee2MQTT fails to start -alpine LXC

### DISCUSSION THREAD

**Discussion by KimCarlsen** [tteck/Proxmox#2254] [Help Needed]
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

Hi. 
Followed the guide on setting up Z2M on Proxmox using Alpine LXC (priviliged)
But i fails to start when rebooting the container.
I can then sign in and start it manually.

Its already set to autostart
z2m:~# rc-update add zigbee2mqtt default
 * rc-update: zigbee2mqtt already installed in runlevel `default'; skipping

How can i troubleshoot?  Note. I'm totally new to LXC and Alpine.
So is there an issue with the scripts? Thanks for great scripts BTW.

thanks
Kim

### Which Linux distribution are you employing?

Default Settings

**Comment by mues-lee** [comment]
with `rc-status` you can see the current status of the service, if it indicates a crash for zigbee2MQTT after the restart, you can take a look at the sys messages to find the cause--> `tail /var/log/messages`

**Reply by KimCarlsen** [reply]
Thanks. it is indeed crashed. however it doesn't show anything in the log. (that i see as an error.)
z2m:~ rc-status
Runlevel: default
 zigbee2mqtt                                                                                                                                      [  crashed  ]
 sysctl                                                                                                                                           [  started  ]
 networking                                                                                                                                       [  started  ]
 crond                                                                                                                                            [  started  ]
Dynamic Runlevel: hotplugged
Dynamic Runlevel: needed/wanted
 localmount                                                                                                                                       [  started  ]
Dynamic Runlevel: manual
z2m:~ tail /var/log/messages
Dec 27 11:29:41 z2m user.debug : Sending signal 15 to PID 394
Dec 27 11:29:41 z2m syslog.info syslogd exiting
Dec 27 11:29:50 z2m syslog.info syslogd started: BusyBox v1.36.1
Dec 27 11:29:50 z2m daemon.info init: starting pid 411, tty '': '/sbin/openrc default'
Dec 27 11:29:50 z2m cron.info crond[463]: crond (busybox 1.36.1) started, log level 8
Dec 27 11:29:50 z2m daemon.info init: starting pid 497, tty '': '/sbin/getty 38400 console'
Dec 27 11:29:50 z2m daemon.info init: starting pid 498, tty '/dev/tty1': '/sbin/getty 38400 tty1'
Dec 27 11:29:50 z2m daemon.info init: starting pid 499, tty '/dev/tty2': '/sbin/getty 38400 tty2'
Dec 27 11:30:00 z2m cron.info crond[463]: USER root pid 516 cmd run-parts /etc/periodic/15min
Dec 27 11:30:19 z2m auth.info login[517]: root login on 'tty1'

**Comment by KCarlsen** [comment]
could it be related to this?
https://github.com/Koenkk/zigbee2mqtt/issues/19869

**Reply by tteck** [reply]
They did mention mixed repos, which the script was changed to here https://github.com/tteck/Proxmox/commit/95e26b68427592dbf25e8f187a0ebe7ab22cce51

To use only edge repositories, execute the command below

```
  cat <<EOF >/etc/apk/repositories
https://dl-cdn.alpinelinux.org/alpine/edge/main
https://dl-cdn.alpinelinux.org/alpine/edge/community
https://dl-cdn.alpinelinux.org/alpine/edge/testing
EOF
```

**Reply by KCarlsen** [reply]
okay, i ran the scripts today as described on your webpage, so its already included.

**Reply by KimCarlsen** [reply]
seems there is more wrong with it. the update does not update Z2M, only OS packages

**Reply by tteck** [reply]
zigbee2mqtt is an Alpine package.

**Reply by tteck** [reply]
https://pkgs.alpinelinux.org/packages?name=zigbee2mqtt&branch=v3.19&repo=&arch=x86_64&maintainer=

**Reply by tteck** [reply]
The Alpine package hasn't been updated.

**Reply by KimCarlsen** [reply]
yes i saw that,my bad. 
I can upgrade by using apk update && apk upgrade  but update.sh is not there.
tried creating a new LXC with this settings:
Using Advanced Settings
Using Container Type: 0
Using Root Password: Automatic Login
Container ID: 101
Using Hostname: zigbee2mqtt
Using Disk Size: 0.3
Allocated Cores: 1
Allocated RAM: 256
Using Bridge: vmbr0
Using IP Address: 192.168.20.42/24
Using Gateway IP Address: 192.168.20.254
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: 192.168.0.254
Using Vlan: 20
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Alpine-Zigbee2MQTT LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using Data for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 101 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.20.42
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.121.3
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Alpine-Zigbee2MQTT
 ✓ Customized Container
 ✓ Completed Successfully!

then in the LXC console:
zigbee2mqtt login: root
Alpine-Zigbee2MQTT LXC provided by https://tteck.github.io/Proxmox/

zigbee2mqtt:~# update
-ash: update: not found

shouldn't update be available directly?

**Reply by tteck** [reply]
Alpine LXCs doesn't have the `update` command

**Reply by KimCarlsen** [reply]
okay makes sense. Perhaps an idea to add to the documentation how to update everything when using Alpine.
but anyway. I'm still stuck with the crashing service, even when creating a new new container., by following the alpine guide it seem like the same issues as #1438, so are all of the fixes implemented?
